Author: Michael Tautschnig <mt@debian.org>
Subject:  Conflicting types for variable ip_optlen
 - globals.h: extern char ip_optlen;
 - main.c: unsigned ip_optlen;
 Undefined behaviour if the value of ip_optlen exceeds 127 for
 any architecture with signed char type.
Bug-Debian: https://bugs.debian.org/688458

Index: hping3.git/main.c
===================================================================
--- hping3.git.orig/main.c
+++ hping3.git/main.c
@@ -152,7 +152,7 @@ unsigned char
 	lsr		[255] = {0},
 	ssr		[255] = {0};
 
-unsigned
+char
 	ip_optlen	= 0;
 
 struct sockaddr_in
